### PR TITLE
fix mapping between C# types and NeoVM types

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo-devpack-dotnet/issues/290
Wait for https://github.com/neo-project/neo/pull/1701

### CODE REVIEW

**Comment by shargon** [comment]
Bool and tuple doesn't return the expected type.

File: tests/Neo.Compiler.MSIL.UnitTests/TestClasses/Contract_Types.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            ret.Value = "neo";
+            return ret;
+        }
+        public static DummyStruct checkStruct()
+        {
+            var ret = new DummyStruct();
+            ret.Value = "neo";
+            return ret;
+        }
+        //public static (string value1, string value2) checkTuple()
```
Comment (shargon) [inline tests/Neo.Compiler.MSIL.UnitTests/TestClasses/Contract_Types.cs:71]: It fail

**Comment by erikzhang** [comment]
@lightszero Can you add tuple support?

**Comment by erikzhang** [comment]
> Bool and tuple doesn't return the expected type.

It seems that there is no `bool` type in MSIL.

**Comment by shargon** [comment]
> It seems that there is no bool type in MSIL.

Yes, so we should change the issue?

`
True:  ldc.i4.1
False: ldc.i4.0
`

**Comment by erikzhang** [comment]
Updated.

**Comment by lightszero** [comment]
I think we support tuple => array  already.
I will check that.

**Comment by shargon** [comment]
@lightszero like this it fail

```
var tuple = ("neo", "smart economy");
            return tuple ;
```

like this it works

```
            return  ("neo", "smart economy");
```

**Comment by shargon** [comment]
`float` works as expected but `Decimal` and `Double` doesn't throw an `unsupported instruction`

**Pull request merged**